gke/autopilot/custom_service_account/main.tf

```diff
@@ -0,0 +1,38 @@
+/**
+* Copyright 2025 Google LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/**
+* Create an Autopilot cluster that uses a custom IAM node service account
+* with the minimum role for GKE.
+* Before you run this sample, create a custom service account by following
+* https://cloud.google.com/kubernetes-engine/docs/how-to/hardening-your-cluster#use_least_privilege_sa
+*/
+
+
+# [START gke_autopilot_custom_service_account]
+resource "google_container_cluster" "autopilot_cluster" {
+  name     = "autopilot-custom-account"
+  location = "us-central1"
+
+  enable_autopilot = true
+
+  cluster_autoscaling {
+    auto_provisioning_defaults {
+      service_account = google_service_account.default.email
```

Comment:
The service account needs to be present for the automated tests, however you can create it outside of any snip.

Reply:
Done, thanks @apeabody